File: freecodecamp/daily-code-challenge/2026-04-03.js
function getBrowserHistory(commands) {
  let history = [];
  let current = -1;

  for (let cmd of commands) {
    if (cmd === "Back") {
      if (current > 0) current--;
    } else if (cmd === "Forward") {
      if (current < history.length - 1) current++;
    } else {
      // It's a URL
      history = history.slice(0, current + 1); // remove forward history
      history.push(cmd);
      current++;
    }
  }

  return [history, current];
}

File: freecodecamp/daily-code-challenge/2026-04-03.py
def get_browser_history(commands):
    history = []
    current = -1

    for cmd in commands:
        if cmd == "Back":
            if current > 0:
                current -= 1
        elif cmd == "Forward":
            if current < len(history) - 1:
                current += 1
        else:
            # URL visit
            history = history[:current + 1]  # remove forward history
            history.append(cmd)
            current += 1

    return [history, current]